internal/api/http.go

```diff
@@ -232,12 +237,20 @@ func newHeaderRoundTripper(authToken string, headers map[string]string, rt http.
 		a += ", application/vnd.github.shadow-cat-preview"
 		headers[accept] = a
 	}
-	return headerRoundTripper{headers: headers, rt: rt}
+	return headerRoundTripper{host: host, headers: headers, rt: rt}
 }
 
 func (hrt headerRoundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
+	// If request is not for the host we have configured, do not add authorization headers.
+	if !isValidHost(req.URL.Host, hrt.host) && hrt.headers[authorization] != "" {
+		delete(hrt.headers, "Authorization")
```

Comment:
```suggestion
		delete(hrt.headers, authorization)
```

Reply:
The key in this map is actually `Authorization`. Applying your suggestion would make the test fail.

map:

```go
map[Accept:application/vnd.github.merge-info-preview+json, application/vnd.github.nebula-preview, application/vnd.github.antiope-preview, application/vnd.github.shadow-cat-preview Authorization:token oauth_token Content-Type:application/json; charset=utf-8 Time-Zone:Europe/London User-Agent:go-gh]
```

failing test:

```go
--- FAIL: TestNewHTTPClientWithDifferentHost (0.00s)
    --- FAIL: TestNewHTTPClientWithDifferentHost/removes_authorization_header_for_a_different_host (0.00s)
        /Users/craigg/code/go-gh/internal/api/http_test.go:145: 
            	Error Trace:	http_test.go:145
            	Error:      	Not equal: 
            	            	expected: ""
            	            	actual  : "token oauth_token"
            	            	
            	            	Diff:
            	            	--- Expected
            	            	+++ Actual
            	            	@@ -1 +1 @@
            	            	-
            	            	+token oauth_token
            	Test:       	TestNewHTTPClientWithDifferentHost/removes_authorization_header_for_a_different_host
FAIL
FAIL	github.com/cli/go-gh/internal/api	0.141s
FAIL
```

Reply:
Ignore that. Completely overlooked that it was a const 😸